tasks/ModelTask.cpp

```diff
@@ -1,198 +0,0 @@
-/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
-//======================================================================================
-// Brazilian Institute of Robotics 
-// Authors: Thomio Watanabe
-// Date: December 2014
-//====================================================================================== 
-// add joints, position and velocity
-//======================================================================================
-#include "ModelTask.hpp"
-//======================================================================================
-gazebo::ModelTask::ModelTask(std::string const& name, TaskCore::TaskState initial_state)
-	: ModelTaskBase(name, initial_state),environment(0)
-{
-}
-//======================================================================================
-gazebo::ModelTask::ModelTask(std::string const& name, RTT::ExecutionEngine* engine, TaskCore::TaskState initial_state)
-	: ModelTaskBase(name, engine, initial_state),environment(0)
-{
-}
-//======================================================================================
-gazebo::ModelTask::~ModelTask()
-{
-	delete joint_port;
-	delete link_port;
-	
-	for(JointPort_V::iterator it = joint_port_list.begin();it != joint_port_list.end(); ++it)
-		delete (it)->first;
-	
-	for(LinkPort_V::iterator it = link_port_list.begin(); it != link_port_list.end(); ++it)
-		delete it->first; 
-	
-	joint_port_list.clear();
-	link_port_list.clear();
-}
-//======================================================================================
-void gazebo::ModelTask::setGazeboModel(physics::WorldPtr _world,  physics::ModelPtr _model, int _environment)
-{
-	world = _world;
-	model = _model;
-	environment = _environment;
-//	sdf = model->GetSDF();
-	
-//	Export Gazebo Joints and Links to Rock-Robotics
-	setJoints();
-	setLinks();
-} 
-//======================================================================================
-void gazebo::ModelTask::setJoints()
-{
-	typedef gazebo::physics::Joint_V Joint_V;
-	Joint_V joints;
-			
-	// Get all joints from a model and creates a Rock component InputPort
-	joints = model->GetJoints();
-	for(Joint_V::iterator joint = joints.begin(); joint != joints.end(); ++joint)
```

Comment:
Sets of joints in Rock are nowadays gathered in a single base::samples::Joints structure. Just do that. You can then avoid creating dynamic ports and instead declare the 'joints' (output) and "cmd_joints" (input) ports in the orogen file